### PR TITLE
Cleanup MojoServices in dart:ui

### PR DESCRIPTION
**Pull request by abarth**
This patch cleans up the way we expose Mojo services into Dart. It also adds a
service provider for "view services," which will evetually contain the raw
keyboard service.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @jason-simmons

File: sky/engine/bindings/mojo_services.cc [hunk 1]
```diff
@@ -20,87 +20,97 @@ MojoServices* GetMojoServices() {
...
-    root_bundle_(root_bundle.Pass()) {
-  if (services_ && services_->services_provided_to_embedder.is_pending()) {
-    services_provided_to_embedder_ = services_->services_provided_to_embedder.Pass();
+    root_bundle_(root_bundle.Pass()),
+    incoming_services_(incoming_services.Pass()) {
+  if (services_ && services_->outgoing_services.is_pending()) {
+    outgoing_services_ = services_->outgoing_services.Pass();
   } else {
-    services_provided_to_embedder_ = GetProxy(&services_from_dart_);
+    outgoing_services_ = GetProxy(&services_from_dart_);
```
Comment (Hixie) [inline sky/engine/bindings/mojo_services.cc]: incoming and outgoing are still pretty confusing terms. it's hard to tell at this point of the code which side of the barrier the code thinks it's on.

File: sky/engine/core/dart/mojo_services.dart [hunk 2]
```diff
@@ -4,8 +4,24 @@
...
+
+  static int takeRootBundle() native "MojoServices_takeRootBundle";
+  static int takeIncomingServices() native "MojoServices_takeIncomingServices";
+  static int takeOutgoingServices() native "MojoServices_takeOutgoingServices";
+  static int takeShell() native "MojoServices_takeShell";
+  static int takeView() native "MojoServices_takeView";
+  static int takeViewServices() native "MojoServices_takeViewServices";
+}
+
+int takeRootBundleHandle() => MojoServices.takeRootBundle();
```
Comment (Hixie) [inline sky/engine/core/dart/mojo_services.dart]: `// TODO(abarth): remove these`, i assume?

File: sky/engine/core/dart/mojo_services.dart [hunk 3]
```diff
@@ -4,8 +4,24 @@
...
-int takeViewHandle() native "takeViewHandle";
+/// Mojo handles provided to the application at startup.
+///
+/// The application can take ownership of these handles by calling the static
+/// "take" functions on this object. Once taken, the application is responsible
+/// for managing the handles.
+class MojoServices {
+  MojoServices._();
+
+  static int takeRootBundle() native "MojoServices_takeRootBundle";
```
Comment (Hixie) [inline sky/engine/core/dart/mojo_services.dart]: would be helpful to have dartdocs for each one here saying wtf they are. I tried to write the docs the other day but couldn't really figure it out.

**Comment by Hixie** [comment]
lgtm

File: sky/engine/core/dart/mojo_services.dart (hunk 3 above)
Comment (abarth) [inline sky/engine/core/dart/mojo_services.dart]: Will do in a later patch.

File: sky/engine/core/dart/mojo_services.dart (hunk 2 above)
Comment (abarth) [inline sky/engine/core/dart/mojo_services.dart]: Yep.

**Pull request merged**